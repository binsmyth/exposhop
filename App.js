import React, { Component } from 'react';
import { Provider } from 'react-redux';
import {createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';//thunk
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { PersistGate } from 'redux-persist/integration/react'

import ProductReducer from './ProductReducer';
import AppNavigator from './AppNavigator';

const persistConfig = {
  key: "root",
  storage
};

const persistedReducer = persistReducer(persistConfig, ProductReducer);

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
    persistedReducer, 
    composeEnhancer (applyMiddleware(thunk))
  );

let persistor = persistStore(store, {manualPersist:true});
persistor.purge();
persistor.pause();

export default class App extends Component {
  render() {
    return (
      <Provider store={ store }>
         <PersistGate loading={null} persistor={persistor}>
          <AppNavigator />
        </PersistGate>
      </Provider>
    );
  }
}