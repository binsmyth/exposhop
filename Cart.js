import React, { Component } from 'react';
import { Platform, Text, View , Button, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import {omit} from './getProductCount';
import { colors, fonts, padding, dimensions } from './styles/base.js';

class Cart extends Component {
  render() {
    let addedItems = Object.entries(omit(this.props.product.addedProducts,"total"))||[];
    let Total = 0;
    return (
      <View style={ styles.container }>
        {addedItems.map((value,index)=>{
          Total = Total + (value[1].count * value[1].price);
          return(
            <View key = { index } >
                <Text style = {{ color: 'red' }}>Product: { value[0] }</Text>
                <Text>Quantity: {value[1].count}</Text>
                <Text>Price: {value[1].price}</Text>
                <Text>Subtotal:{value[1].count * value[1].price}</Text>
            </View>)
          }
        )}
        <Text>Total: {Total}</Text>
      </View>
    );
  }
}

const mapStateToProps=(state) => {
    const { product } = state
    return { product }
};

export default connect(mapStateToProps)(Cart)

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'rgba(255,255,255,0.7)',
    marginTop: 0
  },
  header:{
    fontSize: fonts.lg,
    fontFamily: fonts.primary
  },
  section: {
    color:'red'
  }
})
