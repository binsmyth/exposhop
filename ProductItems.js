import React, {Component} from 'react';
import { Alert, Text, View, TouchableOpacity, Image, Button } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addProduct } from './productActions';
import { fetchProducts } from './fetchProducts';

class titleArrays extends Component {
    constructor(props) {
        super(props);
    }
    componentWillMount() {
        const { fetchProducts } = this.props;
        fetchProducts();
    }
    render(){
        let productItems = this.props.product.product
        if(productItems === undefined){
            productItems = [];
        }

        return (
            <TouchableOpacity>
                {productItems.map((productItems,index)=>(
                      <Button 
                        key = { productItems.id } 
                        title= { productItems.name }
                        onPress= {()=>this.props.addProduct(productItems.id)} 
                        />  
                    ))}
            </TouchableOpacity>
        )
    }
}

const mapStateToProps = (state) =>{
    const { product } = state
    return { product }
};

const mapDispatchToProps = dispatch => (
    bindActionCreators({
        addProduct: addProduct,
        fetchProducts: fetchProducts
    }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(titleArrays)