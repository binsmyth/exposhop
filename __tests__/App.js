import React from 'react';
import App from '../App';
import Cart from '../Cart';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import ProductItems from '../ProductItems';

const mockStore = configureStore([]);
jest.useFakeTimers();
jest.mock('react-native-reanimated', () => require('react-native-reanimated/mock'));
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

test('renders correctly', ()=>{
  const app = renderer.create(<App />);
  const tree = app.toJSON();
  expect (tree).toMatchSnapshot();
});


//Solved using this url: https://www.robinwieruch.de/react-connected-component-test
test('renders correctly', ()=>{
  let store;
  let component;

  const addedProducts = {
    "WordPress Pennant":{
      "count": 1,
      "price": "11.05",
    },
    "total": 1,
  };

  beforeEach(()=>{
    store = mockStore({
      product: {
        addedProducts: addedProducts
      }
    });

    component = renderer.create(
      <Provider store={store}>
        <Cart/>
      </Provider>
    )

    const tree = component.toJSON();
    expect (tree).toMatchSnapshot();
  });
});

test('renders correctly', ()=>{
  let store;
  let component;

  beforeEach(()=>{
    store = mockStore({
      myState: 'sample text',
    });

    component = renderer.create(
      <Provider store={store}>
        <ProductItems />
      </Provider>
    )

    const tree = component.toJSON();
    expect (tree).toMatchSnapshot();
  });
});
