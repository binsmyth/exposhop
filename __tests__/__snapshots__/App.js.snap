// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App correctly 1`] = `
<View
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <View
    importantForAccessibility="yes"
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "opacity": 1,
              },
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "backgroundColor": "#eee",
                  "flex": 1,
                  "flexDirection": "column-reverse",
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  pointerEvents="auto"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "paddingTop": 64,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    accessibilityElementsHidden={false}
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#F5FCFF",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityStates={Array []}
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {},
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#007AFF",
                                  "fontSize": 18,
                                  "padding": 8,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            Open Drawer
                          </Text>
                        </View>
                      </View>
                      <Text>
                        Welcome to Market!
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                pointerEvents="box-none"
                style={
                  Object {
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#a7a7aa",
                        "borderBottomWidth": 0.5,
                        "height": 64,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 20,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "bottom": 0,
                            "flexDirection": "row",
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <View
                          pointerEvents="box-none"
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "bottom": 0,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <Text
                            accessibilityTraits="header"
                            allowFontScaling={false}
                            numberOfLines={1}
                            onLayout={[Function]}
                            style={
                              Object {
                                "color": "rgba(0, 0, 0, 0.9)",
                                "fontSize": 17,
                                "fontWeight": "600",
                                "marginHorizontal": 16,
                                "textAlign": "center",
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.5)",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "opacity": undefined,
            "zIndex": undefined,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.5)",
          },
        ]
      }
    />
  </View>
  <View
    accessibilityViewIsModal={false}
    onLayout={[Function]}
    removeClippedSubviews={false}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "bottom": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "top": 0,
          "width": "80%",
        },
        Object {
          "left": undefined,
        },
        Object {
          "opacity": Object {
            "setValue": [Function],
          },
          "transform": Array [
            Object {
              "translateX": undefined,
            },
          ],
          "zIndex": 0,
        },
        Object {
          "backgroundColor": "#fff",
          "width": 320,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RCTScrollView
        alwaysBounceVertical={false}
      >
        <View>
          <View
            onLayout={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 20,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingVertical": 4,
                  },
                  undefined,
                ]
              }
            >
              <View
                accessibilityLabel="Home"
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(0, 0, 0, .04)",
                      "flexDirection": "row",
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "bold",
                          "margin": 16,
                        },
                        Object {
                          "color": "#2196f3",
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    Home
                  </Text>
                </View>
              </View>
              <View
                accessibilityLabel="Products"
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "flexDirection": "row",
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontWeight": "bold",
                          "margin": 16,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, .87)",
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    Products
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
